flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinatorRestoringTest.java

```diff
@@ -208,7 +211,12 @@ public void testRestoreLatestCheckpointedState() throws Exception {
         tasks.add(jobVertex1);
         tasks.add(jobVertex2);
 
-        assertTrue(coord.restoreLatestCheckpointedStateToAll(tasks, false));
+        // set up the coordinator and validate the initial state
+        final CompletedCheckpointStore secondStore =
+                new EmbeddedCompletedCheckpointStore(1, store.getSuspendedCheckpoints());
+        final CheckpointCoordinator secondCoordinator =
+                coordinatorBuilder.setCompletedCheckpointStore(secondStore).build();
+        assertTrue(secondCoordinator.restoreLatestCheckpointedStateToAll(tasks, false));
```

Comment:
With the change of this commit, the test case could actually be split into two parts:

1) Successful checkpoints are stored in the `CompletedCheckpointStore`.
2) If the `CompletedCheckpointStore` contains a checkpoint, then it is used by the `CheckpointCoordinator` to restore state.

1) might already be covered by some other tests.

Reply:
refactored & split in two "stages"